### ISSUE TITLE
Remove unnecessary constructors

### ISSUE THREAD

**Issue by rakyll** [census-instrumentation/opencensus-go#17]
There are lightweight types such as WindowSlidingTime, WindowSlidingCount, and WindowCumulative that are not complicated enough to require a constructor. Remove NewWindowCumulative and make the zero value useful, remove NewWindowSlidingCount and export fields on WindowSlidingCount, and remove NewWindowSlidingTime and export the fields of WindowSlidingTime.

/cc @hyangah @bogdandrutu

**Comment by rakyll** [comment]
Actually the exported types are never useful, we should instead keep the constructors and unexport the types.

```
func NewWindowCumulative() Window
func NewWindowSlidingCount(count uint64, subSets int) Window
func NewWindowSlidingTime(duration time.Duration, subIntervals int) Window
```

**Comment by acetechnologist** [comment]
The types will still be needed  by the exporters. That is how we can populate zpages and other exporters.

**Comment by rakyll** [comment]
Can you give more info how the zpages and exporters need these types please? These types don't export anything. Will they be just type casted?

**Comment by acetechnologist** [comment]
The data collected *ViewData returns the view along with the data collected (aka Rows).
The View have methods to return the interfaces: Window, Aggregation... The exporter needs to type switch on this interface to determine what Window or Aggregation it is and maybe extract or not data from them.
For example WindowSlidingTime needs to have a mehtod Duration() that returns the duration. This will allow the exporter to know that this data was for the last" 30s, 1mn, 5mn or whatever else.
We need Duration() instead of just exporting the field because we want them to be immutable once created.

This also applies to the interface AggregationValue returned in the Rows. It needs to be type switched to the right AggregationDistributionValue, AggregationCountValue...

**Comment by rakyll** [comment]
Closed via #54 and #55.
